src/TodoList.tsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-lone-blocks */
+// import { useEffect, useRef, useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  visibleTodos: Todo[];
+  deleteTodo: (todoId: number) => void;
+  deletingTodoId: number | null;
+  tempTodo: Todo | null;
+  isLoading: boolean;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  toggleTodo: (todoId: number) => void
+  updateTodoList: (updatedTodoItem: Todo) => void
+  setErrorMessage: (newMessage: string) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  visibleTodos,
+  deleteTodo,
+  deletingTodoId,
+  isLoading,
+  setIsLoading,
+  tempTodo,
+  toggleTodo,
+  updateTodoList,
+  setErrorMessage,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
```

Comment:
Use semantic tags for list `ul` `li`. Don't forget to remove default `li` dots and `ul` paddings